pyreisejl/utility/state.py

```diff
@@ -1,24 +1,68 @@
 from dataclasses import dataclass, field
-from typing import Any, Dict
+from queue import Empty, Queue
+from threading import Thread
+from typing import Any, Dict, List
 
 from pyreisejl.utility.helpers import get_scenario_status
 
 
+class Listener:
+    """Runs in the background to read from stdout/stderr of a long lived
+    process"""
+
+    def __init__(self, stream):
+        self.stream = stream
+        self.queue = Queue()
+        self._start()
+
+    def _start(self):
+        t = Thread(target=self._enqueue_output)
+        t.daemon = True
+        t.start()
+
+    def _enqueue_output(self):
+        for line in iter(self.stream.readline, b""):
```

Comment:
https://docs.python.org/3/library/io.html
"""Note that it’s already possible to iterate on file objects using for line in file: ... without calling file.readlines()."""

From this, could you simplify this?

"""for line in self.stream:"""

Reply:
Yeah that is better, think I will do that

Reply:
Done